cmd/tailscale/cli/web.go

```diff
@@ -246,49 +253,41 @@ func synoAuthn() (string, error) {
 	return strings.TrimSpace(string(out)), nil
 }
 
-func authRedirect(w http.ResponseWriter, r *http.Request) bool {
+func authRedirect(w http.ResponseWriter, r *http.Request) (bool, error) {
 	if distro.Get() == distro.Synology {
 		return synoTokenRedirect(w, r)
 	}
-	return false
+	return false, nil
 }
 
-func synoTokenRedirect(w http.ResponseWriter, r *http.Request) bool {
+func synoTokenRedirect(w http.ResponseWriter, r *http.Request) (bool, error) {
```

Comment:
If you're going to return an error, you don't want to accept a `w` in the first place. You want to return the HTML content to write instead, so you know that `w` is still untouched by the time you go to write the 500 in the caller.